### PR TITLE
Warn about common option misspellings

### PR DESCRIPTION
**Pull request by cincodenada**
This is a proposed fix for #6488, the simplest fix of warning.

### CODE REVIEW

**Review by alexandermirzoyan — APPROVED** [review]
Nice feature! 🔥

File: lib/core/Axios.js [hunk 1]
```diff
@@ -97,6 +97,11 @@ class Axios {
       }
     }
 
+    validator.assertOptions(config, {
+      baseUrl: validators.spelling('baseURL'),
+      withXsrfToken: validators.spelling('withXSRFToken')
+    }, true)
```
Comment (jasonsaayman) [inline lib/core/Axios.js:103]:
```suggestion
    }, true);
```

File: lib/helpers/validator.js [hunk 2]
```diff
@@ -52,6 +52,14 @@ validators.transitional = function transitional(validator, version, message) {
   };
 };
 
+validators.spelling = function spelling(correctSpelling) {
+  return (value, opt) => {
+    // eslint-disable-next-line no-console
+    console.warn(`${opt} is likely a misspelling of ${correctSpelling}`)
+    return true
```
Comment (jasonsaayman) [inline lib/helpers/validator.js:59]:
```suggestion
    console.warn(`${opt} is likely a misspelling of ${correctSpelling}`);
    return true;
```

**Review by jasonsaayman — APPROVED** [review]
code style mainly

**Pull request merged**